src/duration.js

```diff
@@ -197,8 +198,26 @@ export class Duration {
    * @return {Duration}
    */
   static fromObject(obj) {
+    if (obj == null || typeof obj !== 'object') {
+      return Duration.invalid(INVALID_INPUT);
+    }
+
+    const values = Util.normalizeObject(obj, Duration.normalizeUnit, true);
+
+    let hasUnits = false;
+    for (let i = 0, l = orderedUnits.length; i < l; i++) {
+      if (values[orderedUnits[i]]) {
```

Comment:
I think we can just check if `values` has any `ownProperties`, right? `Object.keys(values).length === 0`